src/Swarm/Game/Entity.hs

```diff
@@ -567,7 +567,14 @@ isEmpty = all ((== 0) . fst) . elems
 
 -- | Compute the set of capabilities provided by the devices in an inventory.
 inventoryCapabilities :: Inventory -> Set Capability
-inventoryCapabilities = setOf (to elems . traverse . _2 . entityCapabilities . traverse)
+inventoryCapabilities = Set.unions . map (^. entityCapabilities) . nonzeroEntities
+
+nonzeroEntities :: Inventory -> [Entity]
+nonzeroEntities = map snd . filter ((> 0) . fst) . elems
+
+elemsWithCapability :: Capability -> Inventory -> [Entity]
```

Comment:
It would be good to add Haddock comments to `nonzeroEntities` and `elemsWithCapability`.  In particular the fact that `elemsWithCapability` only returns entities with a nonzero count is not obvious from just the name.

Reply:
Added.